ci-infra/azure/main.tf

```diff
@@ -0,0 +1,90 @@
+resource "azurerm_resource_group" "ci_rg" {
+  name     = "${var.ci_rg}${var.ver}"
+  location = var.location
+}
+
+# Create a container registry that will host the images.
+resource "azurerm_container_registry" "acr" {
+  name                   = "${var.container_registry}${var.ver}"
+  resource_group_name    = azurerm_resource_group.ci_rg.name
+  location               = azurerm_resource_group.ci_rg.location
+  sku                    = "Standard"
+  anonymous_pull_enabled = true
+
+  # TODO: Is there a way to push images without logging in?
+  admin_enabled = true
+}
+
+# Create a user assigned identity
+resource "azurerm_user_assigned_identity" "gh_action_user_identity" {
+  name                = "${var.gh_action_user_identity}${var.ver}"
+  location            = azurerm_resource_group.ci_rg.location
+  resource_group_name = azurerm_resource_group.ci_rg.name
+}
+
+resource "azurerm_federated_identity_credential" "gh_action_federated_credential" {
+  name                = "${var.gh_action_federated_credential}${var.ver}"
+  resource_group_name = azurerm_resource_group.ci_rg.name
+  audience            = ["api://AzureADTokenExchange"]
+  issuer              = "https://token.actions.githubusercontent.com"
+  parent_id           = azurerm_user_assigned_identity.gh_action_user_identity.id
+
+  # TODO: What should be the subject when we are not testing just main but the respective PR?
+  subject = "repo:${var.gh_repo}:ref:refs/heads/main"
```

Comment:
I have to see what subject we can provide for pull request specific code change.

Reply:
Are you thinking in secure it further, like, allowing only jobs triggered by `actor` that belong to the committers group?  Or perhaps we could add additional checkers on the jobs themself, for example, allow to run the e2e jobs only the actors belonging the commiters group.

One interesting thing is that when you have a secret hard-coded on the repository, it is not visible to jobs running on `pull_request` events. @fidencio and I learned it the hard way when setting the workflows to kata containers, which currently they run on `pull_request_target` events (secrets are exposed to `pull_request_target`). I assume that with GitHub's OIDC provider the secret will be exposed to `pull_request` events too.